src/frontends/pytorch/src/op/cat.cpp

```diff
@@ -56,6 +58,38 @@ OutputVector translate_cat_common(const NodeContext& context,
             context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(shape_sliced, axis_const, neg_1, zero));
         return {context.mark_node(std::make_shared<v1::Reshape>(tensor, new_shape, false))};
     }
+
+    const auto first_in_type = list_elems.front().get_element_type();
+    const bool is_mixed_type =
+        list_elems.size() > 1 && (std::any_of(std::next(list_elems.begin()),
+                                              list_elems.end(),
+                                              [&first_in_type](const ov::Output<ov::Node>& input) {
+                                                  return input.get_element_type() != first_in_type ||
+                                                         input.get_element_type() == ov::element::dynamic;
```

Comment:
If dynamic type is among other types we need to assume the worst case - when this type will be different from others, so please change that